core/src/main/java/org/elasticsearch/indices/IndicesService.java

```diff
@@ -606,12 +606,18 @@ public void deleteShardStore(String reason, ShardId shardId, ClusterState cluste
      * @param indexSettings {@code Settings} for the given index
      * @return true if the index can be deleted on this node
      */
-    public boolean canDeleteIndexContents(Index index, Settings indexSettings) {
+    public boolean canDeleteIndexContents(Index index, Settings indexSettings, boolean closed) {
         final IndexServiceInjectorPair indexServiceInjectorPair = this.indices.get(index.name());
         if (IndexMetaData.isOnSharedFilesystem(indexSettings) == false) {
```

Comment:
can't this just be `if (IndexMetaData.isOnSharedFilesystem(indexSettings) == false || closed)`